### PR TITLE
[WiP][Testing] Add backslash check to FileName parameter

### PR DESCRIPTION
**Pull request by dcherniev**
Added check of backslash symbol for FileName field in command SystemRequest.
Added check of backslash symbol for SyncFileName field in command PutFileRequest.

### CODE REVIEW

**Comment by dcherniev** [comment]
Please review: @vlantonov, @AGritsevich, @Kozoriz, @anosach-luxoft, @dtrunov, @VVeremjova

**Comment by dtrunov** [comment]
@dcherniev  please use correct way for describe jira task in description for example  [APPLINK-24743](https://adc.luxoft.com/jira/browse/APPLINK-24743)

**Comment by dcherniev** [comment]
@dtrunov Ok, updated the pull request description.

File: src/components/application_manager/src/commands/mobile/system_request.cc [hunk 1]
```diff
@@ -466,6 +466,13 @@ void SystemRequest::Run() {
     file_name = kSYNC;
   }
 
+  std::size_t backslash_index = file_name.find('\\');
+  if (backslash_index != std::string::npos) {
+    LOG4CXX_ERROR(logger_, "File name contains backslash");
+    SendResponse(false, mobile_apis::Result::INVALID_DATA);
```
Comment (Kozoriz) [inline src/components/application_manager/src/commands/mobile/system_request.cc]: @dcherniev , why you dont send 'info", as in put_file?

Comment (dcherniev) [inline src/components/application_manager/src/commands/mobile/system_request.cc]: @Kozoriz Well the other SendResponse() methods in "system_request" command are sending only 2 parameters, so to correspond with them I removed "info" parameter from my SendResponse() method.

File: src/components/application_manager/src/commands/mobile/put_file_request.cc [hunk 2]
```diff
@@ -108,8 +108,20 @@ void PutFileRequest::Run() {
                  &response_params);
     return;
   }
+
   sync_file_name_ =
       (*message_)[strings::msg_params][strings::sync_file_name].asString();
+
+  std::size_t backslash_index = sync_file_name_.find('\\');
```
Comment (anosach-luxoft) [inline src/components/application_manager/src/commands/mobile/put_file_request.cc]: @dcherniev `const`

File: src/components/application_manager/src/commands/mobile/system_request.cc [hunk 3]
```diff
@@ -466,6 +466,13 @@ void SystemRequest::Run() {
     file_name = kSYNC;
   }
 
+  std::size_t backslash_index = file_name.find('\\');
```
Comment (anosach-luxoft) [inline src/components/application_manager/src/commands/mobile/system_request.cc]: @dcherniev `const`

File: src/components/application_manager/src/commands/mobile/put_file_request.cc (hunk 2 above)
Comment (dcherniev) [inline src/components/application_manager/src/commands/mobile/put_file_request.cc]: @anosach-luxoft Fixed, please check 5c5006a

File: src/components/application_manager/src/commands/mobile/system_request.cc (hunk 3 above)
Comment (dcherniev) [inline src/components/application_manager/src/commands/mobile/system_request.cc]: @anosach-luxoft Fixed, please check 5c5006a

**Comment by dcherniev** [comment]
Rebuild required

**Comment by dcherniev** [comment]
Rebuild required

**Comment by dcherniev** [comment]
Rebuild required

**Comment by LuxoftAKutsan** [comment]
Not Actual

**Pull request closed without merging**